auto_submit/lib/model/auto_submit_query_result.dart

```diff
@@ -178,3 +180,35 @@ class QueryResult {
 
   Map<String, dynamic> toJson() => _$QueryResultToJson(this);
 }
+
+@JsonSerializable()
+class OuterRevertPullRequest {
+  OuterRevertPullRequest({this.revertPullRequest});
+
+  @JsonKey(name: 'revertPullRequest')
+  RevertPullRequest? revertPullRequest;
+
+  factory OuterRevertPullRequest.fromJson(Map<String, dynamic> json) => _$OuterRevertPullRequestFromJson(json);
+
+  Map<String, dynamic> toJson() => _$OuterRevertPullRequestToJson(this);
+}
+
+@JsonSerializable()
+class RevertPullRequest {
+  RevertPullRequest({
+    this.clientMutationId,
+    this.pullRequest,
+    this.revertPullRequest,
```

Comment:
I don't quite follow these names. The class name is `RevertPullRequest`, and it has an arg named `revertPullRequest`. I would expect `revertPullRequest` is an instance of `RevertPullRequest`.

Could we use more explicit names? Also what is `revertPullRequest` different from `pullRequest`? Some docs would be helpful.

Reply:
I did post the doc in the description. These are the names that GitHub gives the fields. I kept them the same so they match the documentation.
https://docs.github.com/en/graphql/reference/mutations#revertpullrequest
The overuse the revertPullRequest identifier so this leads to some oddities when decoding the object json that is returned.

Here is an example of this in practice:
The Mutation input:
```
mutation RevertPullFlutterPullRequest ($pullRequestId:ID!) {
  revertPullRequest (
    input: { 
    	body:"Testing revert mutation", 
      clientMutationId: "ra186026", 
      draft: false, 
      pullRequestId: $pullRequestId, 
      title: "Revert comment in configuration file." }
  ) {
    clientMutationId 
    pullRequest {
      author {
        login
      }
      authorAssociation
      id
      title
      number
      body
      repository {
        owner {
          login
        }
        name
      }
    }
    revertPullRequest {
      author {
        login
      }
      authorAssociation
      id
      title
      number
      body
      repository {
        owner {
          login
        }
        name
      }
    }
  }
}
```

Output from the graphql mutation:
```
{
  "data": {
    "revertPullRequest": {
      "clientMutationId": "ra186026",
      "pullRequest": {
        "author": {
          "login": "ricardoamador"
        },
        "authorAssociation": "OWNER",
        "id": "PR_kwDOIRxr_M5MQ7mV",
        "title": "Adding a TODO comment for testing pull request auto approval.",
        "number": 18,
        "body": "",
        "repository": {
          "owner": {
            "login": "ricardoamador"
          },
          "name": "flutter_test"
        }
      },
      "revertPullRequest": {
        "author": {
          "login": "ricardoamador"
        },
        "authorAssociation": "OWNER",
        "id": "PR_kwDOIRxr_M5QN0kD",
        "title": "Revert comment in configuration file.",
        "number": 23,
        "body": "Testing revert mutation",
        "repository": {
          "owner": {
            "login": "ricardoamador"
          },
          "name": "flutter_test"
        }
      }
    }
  }
}
```

Reply:
The pullRequest is the original request that is being reverted.
The revertPullRequest is the newly created pull request that reverts pullRequest.
And revertPullRequest is the name of the mutation.
GitHub's naming conventions I am afraid.

Reply:
Okay I updated the docs in the query file.